src/v/cluster_link/replication/deps.h

```diff
@@ -49,6 +49,13 @@ class data_sink {
 
     // Returns the HWM of the partition
     virtual kafka::offset high_watermark() const = 0;
+
+    // Performs a prefix truncation on the sink partition
+    virtual ss::future<std::error_code> prefix_truncate(
+      kafka::offset truncation_offset, ss::lowres_clock::time_point deadline)
+      = 0;
+
```

Comment:
I'm wondering if this should just be a property of the implementation vs putting it in the API.